packages/indexer-cli/src/commands/indexer/cost/delete.ts

```diff
@@ -0,0 +1,72 @@
+import { GluegunToolbox } from 'gluegun'
+import chalk from 'chalk'
+
+import { loadValidatedConfig } from '../../../config'
+import { createIndexerManagementClient } from '../../../client'
+import { fixParameters } from '../../../command-helpers'
+import { costModels, deleteCostModels, parseDeploymentID } from '../../../cost'
+
+const HELP = `
+${chalk.bold('graph indexer cost delete')} [options] all
+${chalk.bold('graph indexer cost delete')} [options] global
+${chalk.bold('graph indexer cost delete')} [options] <deployment-id>
+
+${chalk.dim('Options:')}
+
+  -h, --help                    Show usage information
+  -o, --output table|json|yaml  Choose the output format: table (default), JSON, or YAML
+`
+
+module.exports = {
+  name: 'delete',
+  alias: [],
+  description: 'Remove one or many cost models',
+  run: async (toolbox: GluegunToolbox) => {
+    const { print, parameters } = toolbox
+
+    const { h, help, o, output } = parameters.options
+    const [rawDeployment] = fixParameters(parameters, { h, help }) || []
+    const outputFormat = o || output || 'table'
+
+    if (help || h) {
+      print.info(HELP)
+      return
+    }
+
+    if (!['json', 'yaml', 'table'].includes(outputFormat)) {
+      print.error(`Invalid output format "${outputFormat}"`)
+      process.exitCode = 1
+      return
+    }
+
+    try {
+      const config = loadValidatedConfig()
+      // Create indexer API client
+      const client = await createIndexerManagementClient({ url: config.api })
+      const deployment = parseDeploymentID(rawDeployment)
+      switch (deployment) {
+        case 'all': {
+          const deployments: string[] = (await costModels(client)).map(
+            model => model.deployment?.toString() ?? '',
```

Comment:
I don't think there should be cost model with `null`, since `costModelAttributes` type requires `deployment` field and CLI validates the deployment id. 
Although graphQL parsing returns `Partial<CostModelAttributes>` so this was a quick way to get around it where we shouldn't have any '' in the deployments array - reviewing now, I think a filter might make more sense

Reply:
Either way works for me.  Probably easiest to simply filter.

Reply:
Updated to enforce type `CostModelAttributes` which require cost model id and deployment but allow null model and variables, with specific (perhaps redundant) filter check on `deployment`